.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  push:
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run unit tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests.sh
+  api-tests:
+    name: API Tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run API tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests-api.sh
+  psr2-check:
+    name: PSR2 check
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Check PSR2
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./check-psr2.sh
+
+  build-and-publish:
+    name: Build and Publish
+    needs: [unit-tests,api-tests,psr2-check]
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ vars.DOCKER_ORG }}/email-service
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./Dockerfile
```

Comment:
This is the default, so you don't really need to include it. See https://github.com/docker/build-push-action.
```suggestion
```